app/controllers/katello/api/registry/registry_proxies_controller.rb

```diff
@@ -258,42 +233,37 @@ def status_upload_blob
       render plain: '', status: :no_content
     end
 
-    def chunk_upload_blob
-      response.header['Location'] = "#{request_url}/v2/#{params[:repository]}/blobs/uploads/#{params[:uuid]}"
-      render plain: '', status: :accepted
+    def translated_headers_for_proxy
+      current_headers = {}
+      env = request.env.select do |key, _value|
+        key.match("^HTTP_.*")
+      end
+      env.each do |header|
+        current_headers[header[0].split('_')[1..-1].join('-')] = header[1]
```

Comment:
@ianballou : Could you explain why this is needed?

Reply:
This is one bit of the code that was created a while back. I believe it's here because request.env contains environment variables, and some of which translate to headers. The headers as env vars have underscores and are prefixed with "HTTP_".  To turn them into true headers, they need the underscores turned into hyphens also the HTTP prefix needs removing. 

I need to read up more on it to verify, but one example I saw was that our apache must set the Remote-User header, which eventually reaches Pulp/Django as "HTTP_REMOTE_USER".

Reply:
I was trying to remember why `start_upload_blob` doesn't use `translated_headers_for_proxy`, and remembered that uploading a blob doesn't require special headers, unlike actually uploading the blob. For example: https://specs.opencontainers.org/distribution-spec/?v=v1.0.0#DISTRIBUTION-SPEC-93

Reply:
Some more helpful material about the web server environment variables https://medium.com/@cipeinado/unlock-http-info-in-rails-with-rack-request-env-and-uri-c7835397dc14
It clarifies my thinking about how the HTTP_ variables relate to headers.

Reply:
Another thing that's interesting -- so `request.headers` doesn't list anything by default. You need to look up a specific header to get info. If you type `request.headers.to_a`, for example, you end up just getting the normal `request.env` information more or less. It would need to be verified, but I think `request.headers['key']` looks for headers by various naming conventions. For example:

```ruby
[13] pry(#<Katello::Api::Registry::RegistryProxiesController>)> request.headers
=> #<ActionDispatch::Http::Headers:0x000055afdecc0450
 @req=
  #<ActionDispatch::Request POST "https://centos8-katello-devel.manicotto.example.com/v2/organization/product/hello-world/blobs/uploads/" for 192.168.122.147>>
[14] pry(#<Katello::Api::Registry::RegistryProxiesController>)> request.headers['Authorization']
=> "Bearer $2a$09$1b6453892473a467d0737uL4crR75S3rqz44qQcdMGK68gYLHX/Ba"
[15] pry(#<Katello::Api::Registry::RegistryProxiesController>)> request.headers['AUTHORIZATION']
=> "Bearer $2a$09$1b6453892473a467d0737uL4crR75S3rqz44qQcdMGK68gYLHX/Ba"
[16] pry(#<Katello::Api::Registry::RegistryProxiesController>)> request.headers['HTTP_AUTHORIZATION']
=> "Bearer $2a$09$1b6453892473a467d0737uL4crR75S3rqz44qQcdMGK68gYLHX/Ba"
[17] pry(#<Katello::Api::Registry::RegistryProxiesController>)> request.headers['HTT_AUTHORIZATION']
=> nil
[18] pry(#<Katello::Api::Registry::RegistryProxiesController>)> request.headers['HTTP_AUTHORIZATION']
=> "Bearer $2a$09$1b6453892473a467d0737uL4crR75S3rqz44qQcdMGK68gYLHX/Ba"
[19] pry(#<Katello::Api::Registry::RegistryProxiesController>)> request.headers['x-authorization']
=> nil
[20] pry(#<Katello::Api::Registry::RegistryProxiesController>)> request.headers['X-Authorization']
=> nil
```

Reply:
I did realize that some headers, like the Content-Length, doesn't make it in with the HTTP_ env vars. So I'll add explicit checks for those.